components/template-mgt/org.wso2.carbon.identity.template.mgt/src/main/java/org/wso2/carbon/identity/template/mgt/handler/impl/FileBasedTemplateHandler.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.template.mgt.handler.impl;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.configuration.mgt.core.search.Condition;
+import org.wso2.carbon.identity.template.mgt.exception.TemplateManagementClientException;
+import org.wso2.carbon.identity.template.mgt.exception.TemplateManagementException;
+import org.wso2.carbon.identity.template.mgt.handler.ReadOnlyTemplateHandler;
+import org.wso2.carbon.identity.template.mgt.model.Template;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.wso2.carbon.identity.template.mgt.TemplateMgtConstants.ErrorMessages.ERROR_CODE_INVALID_ARGUMENTS_FOR_LIMIT;
+import static org.wso2.carbon.identity.template.mgt.TemplateMgtConstants.ErrorMessages.ERROR_CODE_INVALID_ARGUMENTS_FOR_OFFSET;
+import static org.wso2.carbon.identity.template.mgt.internal.TemplateManagerDataHolder.getFileBasedTemplates;
+import static org.wso2.carbon.identity.template.mgt.util.TemplateMgtUtils.handleClientException;
+
+/**
+ * File based template handler implementation.
+ */
+public class FileBasedTemplateHandler implements ReadOnlyTemplateHandler {
+
+    private static final Log log = LogFactory.getLog(FileBasedTemplateHandler.class);
+    private static final Integer DEFAULT_SEARCH_LIMIT = 100;
+
+    @Override
+    public Template getTemplateById(String templateId) throws TemplateManagementException {
+
+        return getFileBasedTemplates().get(templateId);
+    }
+
+    @Override
+    public List<Template> listTemplates(String templateType, Integer limit, Integer offset, Condition
+            searchCondition) throws TemplateManagementException {
+
+        validatePaginationParameters(limit, offset);
+
+        if (limit == 0) {
+            limit = DEFAULT_SEARCH_LIMIT;
```

Comment:
there's a config in identity.xml file for default limit. Let's use that

Reply:
https://github.com/wso2/carbon-identity-framework/blob/398d82dfd77afc51611e43e378909d74851d6138/components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityUtil.java#L1419

Reply:
Basically use IdentityUtil.getDefaultItemsPerPage() instead of DEFAULT_SEARCH_LIMIT